src/Routes/DynamicRoute/DynamicRoute.tsx

```diff
@@ -20,12 +20,15 @@ const DynamicRoute: React.FC<DynamicRouteProps> = ({ location }) => {
   const dynamicRoutePages = useExtensions<DynamicRoutePage>(isDynamicRoutePage);
   React.useEffect(() => {
     if (location) {
-      const [, , app] = location.pathname?.split('/') || [];
-      if (app) {
+      const [, first, second] = location.pathname?.split('/') || [];
+      if (first || second) {
         const { properties: currRoute, pluginName } =
-          dynamicRoutePages.find(({ properties }) =>
-            Array.isArray(properties.path) ? properties.path.includes(`/${app}`) : properties.path === `/${app}`,
-          ) || {};
+          dynamicRoutePages.find(({ properties }) => {
+            if (Array.isArray(properties.path)) {
+              return properties.path.includes(`/${second}`) || properties.path.includes(`/${first}`)
+            }
+            return properties.path === `/${second}` || properties.path === `/${first}`
```

Comment:
Mhh, instead of some complex logic we might want to use https://v5.reactrouter.com/web/api/matchPath to pick the right component. The params `exact` and `strict` are part of the route, right?

Reply:
We have `exact` as an [optional prop in the extension](https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/extensions/pages.ts#L28). But there will still be `:something` variables we need to parse. I do not believe `exact` means equality here, I think it means number of URL parts are equal.